Adding support for histograms

lcavadas opened the pull request:
- Added a new HttpHistogramMeterFilterFactory class to contribute to the configuration of the 'http.server.requests' and 'http.client.requests' meters.

- Added test to validate configuration takes effect.

lcavadas committed:
Adding support for histograms

- Added a new HttpHistogramMeterFilterFactory class to contribute to the configuration of the 'http.server.requests' and 'http.client.requests' meters.

- Renamed the existing HttpMeterFilterFactory to HttpPercentilesMeterFilterFactory in order for the naming of both classes to be consistent.

- Added test to validate configuration takes effect.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/micronaut-projects/micronaut-micrometer?pullRequest=740) <br/>All committers have signed the CLA.

sdelamo commented on micrometer-core/src/main/java/io/micronaut/configuration/metrics/binder/web/HttpPercentilesMeterFilterFactory.java:
```diff
@@ -39,43 +37,46 @@
 @RequiresMetrics
 @Requires(property = WebMetricsPublisher.ENABLED, notEquals = FALSE)
-public class HttpMeterFilterFactory {
+public class HttpPercentilesMeterFilterFactory {
```
This is a breaking change. We cannot rename public classes in a patch version.

lcavadas replied:
Done, classname back to original 👍

lcavadas committed:
CR Fix

Leave the original percentiles class alone so this PR can be released as a patch

lcavadas committed:
CR Fix

Leave the original percentiles class alone so this PR can be released as a patch

graemerocher commented:
Thanks for the contribution!

scprek reviewed (commented):
nvm docs are updated.
